### ISSUE TITLE
Set request initiator origin  in access-control-allow-origin header to solve CORS issue

### ISSUE THREAD

**Issue by lazyvab** [requestly/requestly#622]
To bypass CORS issue for a request in browser, `access-control-allow-origin` response header can be set as `*` using `Modify Headers` rule. But it does not work in some cases like when the request's credentials mode is 'include'.

Looking for a solution where the domain of the page can be populated as the header value, i.e.
```
When request URL contains "localhost",
Add response header "Access-Control-Allow-Origin" = {{page_domain}}
```

![image](https://github.com/requestly/requestly/assets/6367566/46391467-fe1d-4e75-90c5-8ee1efcd43c9)

**Comment by lazyvab** [comment]
In case a request originates from an iframe, the value of `access-control-allow-origin` response header should be origin derived from the iframe URL, instead of main page URL.

So, going to expose a new predefined function `rq_request_initiator_origin()` which can be used to get desired value.

**Comment by lazyvab** [comment]
Released in extension version `23.5.19`.
